File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/modloaders/FabricVersion.java
package net.kdt.pojavlaunch.modloaders;

import androidx.annotation.NonNull;

public class FabricVersion {
    public String version;
    public boolean stable;

    @NonNull
    @Override
    public String toString() {
        return version;
    }

    public static class LoaderDescriptor extends FabricVersion {
        public FabricVersion loader;

        @NonNull
        @Override
        public String toString() {
            return loader != null ? loader.toString() : "null";
        }
    }
}

File: app_pojavlauncher/src/main/java/net/kdt/pojavlaunch/modloaders/FabricVersion.kt
package net.kdt.pojavlaunch.modloaders

open class FabricVersion {
    var version: String? = null
    var stable: Boolean = false

    class LoaderDescriptor : FabricVersion() {
        var loader: FabricVersion? = null

        override fun toString(): String = loader?.toString() ?: "null"
    }

    override fun toString(): String = version ?: ""
}
